hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -900,8 +904,8 @@ def integer_array_indices(
     shape: Shape,
     *,
     result_shape: st.SearchStrategy[Shape] = array_shapes(),
-    dtype: np.dtype = "int",
-) -> st.SearchStrategy[Tuple[np.ndarray, ...]]:
+    dtype: D = np.int32,
```

Comment:
I think this default value is also a change from the current behavior, which iirc resolves to int32 or int64 depending on 32/64-bit systems.

Reply:
Sure, I checked what `dtype("int")` gave back, but your are right that depends on the architecture.

The problem with strings as dtypes is that they keep the type checker from being able to infer the correct dtype.

I think `np.int_` should have the same behaviour as "int", I'll change it to that.